### DISCUSSION TITLE
Adjusting RGB Colors of an image?

### DISCUSSION THREAD

**Discussion by biggrizzleyguy** [SixLabors/ImageSharp#1838] [Q&A] [answered]
Is there a way to adjust the RGB values of an entire image?  I'd like to be able to use sliders to add some red, or green to the overall color of the image.    
![image](https://user-images.githubusercontent.com/13790656/142571205-46cd050e-8e4b-4058-a0cd-2b88307a74aa.png)

The ultimate goal is to be able to read a 50% grey card and get the average values and be able to adjust them. Below is just an example of reading an area of the grey card and finding it's average RGB values.  I'd like to add some RGB color sliders to this, or possibly automate an adjustment to change the color of the image to reflect an average of 128,128,128. (I realize the white crosshairs are going to skew the RGB average a bit).  Then take those same adjustments and apply them to other images shot at the same photo shoot.
![image](https://user-images.githubusercontent.com/13790656/142571776-fba481b3-7359-4ad9-ae93-600ff9c696bf.png)

Any help is greatly appreciated!

**Comment by antonfirsov** [comment, marked as answer]
I'm not aware of a built-in processor for this, but I think you should be able to implement this with, `image.Mutate(c => c.Filter(new ColorMatrix( .... )))`. You need to figure out how to create a `ColorMatrix` from your R,G,B slider values.

Some ideas from another library with equivalent feature:
https://docs.rainmeter.net/tips/colormatrix-guide/

**Reply by JimBobSquarePants** [reply]
Yep Colormatrix is definitely the way to go.

**Reply by biggrizzleyguy** [reply]
Thank you for steering me in the right direction.  I'll need to learn more about Colormatrix.
